product_code/user_service/impl/src/test/scala/de/upb/cs/uc4/user/impl/UserSpec.scala

```diff
@@ -7,62 +7,209 @@ import org.scalatest.wordspec.AnyWordSpecLike
+      errorVariables should contain theSameElementsAs Seq("birthDate")
+    }
+    "return a validation error for incorrect format in birthDate" in {
+      val errors = adminValid.copy(birthDate = "2019/1/1").validate
+      val errorVariables = errors.map(error => error.name)
+      errorVariables should contain theSameElementsAs Seq("birthDate")
     }
   }
 
   "A Lecturer" should {
     "be validated" in {
-      val errors = lecturer0.validate
+      val errors = lecturerValid.validate
       errors shouldBe empty
     }
-    "return validation errors" in {
-      val errors = lecturer1.validate
+
+    //FREE TEXT
+    "return a validation error for incorrect length in freeText" in {
+      val errors = lecturerValid.copy(freeText = genericString * 101).validate
+      val errorVariables = errors.map(error => error.name)
+      errorVariables should contain theSameElementsAs Seq("freeText")
+    }
+
+    //RESEARCH AREA
+    "return a validation error for incorrect length in researchArea" in {
+      val errors = lecturerValid.copy(researchArea = genericString * 3).validate
       val errorVariables = errors.map(error => error.name)
-      errorVariables should contain only ("role", "email", "phoneNumber", "birthDate", "firstName", "lastName")
+      errorVariables should contain theSameElementsAs Seq("researchArea")
     }
   }
 
   "A Student" should {
     "be validated" in {
-      val errors = student0.validate
+      val errors = studentValid.validate
+      errors shouldBe empty
+    }
+
+    //LATEST IMMATRICULATION
+    "be validated with empty latestImmatriculation" in {
+      val errors = studentValid.copy(latestImmatriculation = "").validate
       errors shouldBe empty
     }
-    "return validation errors" in {
-      val errors = student1.validate
+    "return a validation error for incorrect semester in latestImmatriculation" in {
+      val errors = studentValid.copy(latestImmatriculation = "FS2020").validate
+      val errorVariables = errors.map(error => error.name)
+      errorVariables should contain theSameElementsAs Seq("latestImmatriculation")
+    }
+
+    //MATRICULATION ID
+    "return a validation error for incorrect format in matriculationId" in {
+      val errors = studentValid.copy(matriculationId = "0000000").validate
+      val errorVariables = errors.map(error => error.name)
+      errorVariables should contain theSameElementsAs Seq("matriculationId")
+    }
+    "return a validation error for incorrect length in matriculationId" in {
+      val errors = studentValid.copy(matriculationId = "00042").validate
       val errorVariables = errors.map(error => error.name)
-      errorVariables should contain only ("role", "email", "phoneNumber", "birthDate", "firstName", "lastName", "matriculationId", "latestImmatriculation")
+      errorVariables should contain theSameElementsAs Seq("matriculationId")
     }
   }
 
-  "An Adress" should {
+  "An Address" should {
```

Comment:
Address could be checked in its own Spec (but then every UserType could be checked in its own Spec (maybe would be better)).

Reply:
I agree, we COULD do that, but I dont really see the reason to do that now. We may do that if the file gets too large (which may very well be in this sprint, if we write more tests for users and the address)